pytorch_lightning/utilities/auto_restart.py

```diff
@@ -571,3 +571,18 @@ def reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, A
 
     else:
         raise MisconfigurationException("This shouldn't happen. Please, open an issue on PyTorch Lightning Github.")
+
+
+def _detect_fault_tolerant_training_mode() -> _FaultTolerantTrainingMode:
```

Comment:
This should be a classmethod in `_FaultTolerantTrainingMode`

Reply:
Done !